notifier/notifier.go

```diff
@@ -0,0 +1,103 @@
+package notifier
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+)
+
+const (
+	// IOS identifier for an ios mnotification.
+	IOS = 1
+	// Android identifier for an android notification.
+	Android = 2
+)
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Notifier handles android and ios push notifications.
+type Notifier struct {
+	client httpClient
+	url    string
+}
+
+// New notifier connected to the specified server.
+func New(url string) *Notifier {
+	client := &http.Client{}
+
+	return &Notifier{url: url, client: client}
+}
+
+// Notification details for gorush.
+type Notification struct {
+	Tokens   []string `json:"tokens"`
+	Platform float32  `json:"platform"`
+	Message  string   `json:"message"`
+}
+
+type request struct {
+	Notifications []*Notification `json:"notifications"`
+}
+
+// Response from gorush.
+type Response struct {
+	Logs []struct {
+		Type  string `json:"type"`
+		Error string `json:"error"`
+	} `json:"logs"`
+}
+
+// Send a push notification to given devices.
+func (n *Notifier) Send(notifications []*Notification) error {
+	url := n.url + "/api/push"
+	r := request{Notifications: notifications}
+
+	body, err := json.Marshal(r)
+	if err != nil {
+		return err
+	}
+
+	res, err := n.doRequest(url, body)
+	if err != nil {
+		return err
+	}
+
+	if len(res.Logs) > 0 {
+		if res.Logs[0].Type == "failed-push" {
+			return errors.New(res.Logs[0].Error)
```

Comment:
This may be cryptic. Maybe `fmt.Errorf("notifier error: %v", res.Logs[0].Error)`?